src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -668,6 +697,46 @@ string GetSeverity(LogLevel level)
                         WriteMetadata(MetadataKeys.Severity, nameof(LogLevel.Error));
                     }
                 }
+                
+                if (!string.IsNullOrEmpty(_task.ImplicitPlatformPackageIdentifier) &&
+                    !string.IsNullOrEmpty(_task.ExpectedPlatformPackageVersion) &&
+                    //  If RuntimeFrameworkVersion was specified as a version range or a floating version,
+                    //  then we can't compare the versions directly, so just skip the check
+                    _task.ExpectedPlatformPackageVersion.IndexOfAny(_specialNuGetVersionChars) < 0)
+                {
+                    var platformLibrary = _runtimeTarget.GetLibrary(_task.ImplicitPlatformPackageIdentifier);
+                    if (platformLibrary != null)
+                    {
+                        bool HasTwoPeriods(string s)
+                        {
+                            int firstPeriodIndex = s.IndexOf('.');
+                            if (firstPeriodIndex < 0)
+                            {
+                                return false;
+                            }
+                            int secondPeriodIndex = s.IndexOf('.', firstPeriodIndex + 1);
+                            return secondPeriodIndex >= 0;
+                        }
+
+                        string restoredPlatformLibraryVersion = platformLibrary.Version.ToNormalizedString();
+
+                        //  Normalize expected version.  For example, converts "2.0" to "2.0.0"
+                        string expectedPlatformPackageVersion = _task.ExpectedPlatformPackageVersion;
+                        if (!HasTwoPeriods(expectedPlatformPackageVersion))
+                        {
+                            expectedPlatformPackageVersion += ".0";
+                        }                        
+
+                        if (restoredPlatformLibraryVersion != expectedPlatformPackageVersion)
+                        {
+                            WriteItem(string.Format(Strings.MismatchedPlatformPackageVersion,
+                                                    _task.ImplicitPlatformPackageIdentifier,
+                                                    restoredPlatformLibraryVersion,
+                                                    expectedPlatformPackageVersion));
+                            WriteMetadata(MetadataKeys.Severity, nameof(LogLevel.Error));
+                        }
```

Comment:
I like to see this refactored a bit so that I can quickly read WriteAdditionalLogMessages and see which issues these are. Right now it's a little buried that all of this is for one specific additional message.

Reply:
@nguerrera I refactored this:

```c#
private void WriteAdditionalLogMessages()
{
    WriteUnsupportedRuntimeIdentifierMessageIfNecessary();
    WriteMismatchedPlatformPackageVersionMessageIfNecessary();
}

```

Reply:
Thanks. 👍